File: src/org/usfirst/frc5293/commands/teleop/events/CameraSetOriginOnPressed.java
package org.usfirst.frc5293.commands.teleop.events;

import org.usfirst.frc5293.Subsystems;
import org.usfirst.frc5293.commands.util.ActionCommand;
import org.usfirst.frc5293.translations.camera.CameraEngine;
import org.usfirst.frc5293.translations.util.Point;

public class CameraSetOriginOnPressed extends ActionCommand {

    public CameraSetOriginOnPressed() {
        super(Subsystems.getCamera());
    }

    @Override
    protected void action() {
        Point point = CameraEngine.getInstance().getResult();
        Subsystems.getCamera().setOrigin(point.x, point.y);
    }
}

File: src/main/kotlin/org/usfirst/frc5293/commands/teleop/events/CameraSetOriginOnPressed.kt
package org.usfirst.frc5293.commands.teleop.events

import org.usfirst.frc5293.Subsystems
import org.usfirst.frc5293.commands.util.SubsystemCommand
import org.usfirst.frc5293.translations.camera.CameraEngine

class CameraSetOriginOnPressed : SubsystemCommand(Subsystems.camera) {

    override fun action() {
        val point = CameraEngine.result
        Subsystems.camera.setOrigin(point.x, point.y)
    }
}
